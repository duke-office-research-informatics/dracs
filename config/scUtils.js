import React from 'react';
import { ThemeProvider } from 'styled-components';
import { mount, shallow } from 'enzyme';
import { theme } from '../lib/dracs.es.js';

export const shallowWithTheme = (tree) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();
  return shallow(tree, { context });
};

export const mountWithTheme = (tree) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext();

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes // Needed so child components receive the theme prop
  });
};

//FOR ACCESSING PORTALS VIA ENZYME -- import portal from dracs
// https://github.com/airbnb/enzyme/issues/1202#issuecomment-351639783
// function mountPortal(el){
//   const wrapper =  mountWithTheme(el);
//   wrapper.setState({ visible: true});
//   const portalInstance = wrapper.find(Portal).instance();
//   return mountWithTheme(portalInstance.props.children);
// }

//FOR ACCESSING EVENT LISTENERS ATTACHED TO DOCUMENT OR DOCUMENT.BODY
//https://github.com/airbnb/enzyme/issues/426#issuecomment-296304015
// const map = {};
// document.body.addEventListener = jest.fn((event, cb) => {
//   map[event] = cb;
// });
