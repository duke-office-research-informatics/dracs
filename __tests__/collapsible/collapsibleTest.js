import React from "react";
import { mountWithTheme } from "../../config/scUtils.js";
import { axe } from "jest-axe";
import { theme, Collapsible, IconMenu, IconHelp } from "../../lib/dracs.es.js";

const mockFn = jest.fn();

describe("Collapsible", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mountWithTheme(
      <Collapsible title="hello">
        <IconMenu />
      </Collapsible>
    );
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("returns no a11y violations", async () => {
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
  });

  it("matches snapshot and returns no a11y violations when active is set to true", async () => {
    wrapper.setProps({ active: true });
    const results = await axe(wrapper.html());
    expect(results).toHaveNoViolations();
    expect(wrapper).toMatchSnapshot();
  });

  it("Does not display body when state.active is false", () => {
    expect(wrapper.state().active).toBeFalsy();
    expect(wrapper.find("collapsible__BodyWrap")).toHaveStyleRule(
      "display",
      "none"
    );
  });

  it("Displays body and sets state.active to true on toggle click", () => {
    expect(wrapper.state().active).toBeFalsy();
    wrapper.find("collapsible__CollapseToggle").simulate("click");
    expect(wrapper.find("collapsible__BodyWrap")).toHaveStyleRule(
      "display",
      "flex"
    );
    expect(wrapper.state().active).toBe(true);
  });

  it("Displays body and sets state.active to true on title click", () => {
    expect(wrapper.state().active).toBeFalsy();
    wrapper.find("collapsible__ToggleWrap").simulate("click");
    expect(wrapper.find("collapsible__BodyWrap")).toHaveStyleRule(
      "display",
      "flex"
    );
    expect(wrapper.state().active).toBe(true);
  });

  it("Hides body and sets state.active to false when toggle is clicked and active is true", () => {
    wrapper.setProps({ active: true });
    expect(wrapper.state().active).toBe(true);
    wrapper.find("collapsible__ToggleWrap").simulate("click");
    expect(wrapper.state().active).toBe(false);
    expect(wrapper.find("collapsible__BodyWrap")).toHaveStyleRule(
      "display",
      "none"
    );
  });

  it("sets wrapper padding via props", () => {
    wrapper.setProps({ componentWrapperPadding: "8px" });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("collapsible__ElementWrap")).toHaveStyleRule(
      "padding",
      "8px"
    );
  });

  it("passes click function to toggle via onToggleClick prop", () => {
    wrapper.setProps({ onToggleClick: mockFn });
    expect(wrapper).toMatchSnapshot();
    wrapper.find("collapsible__ToggleWrap").simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });

  it("passes a string title via props", () => {
    expect(wrapper.find("collapsible__ToggleWrap").text()).toContain("hello");
  });

  it("passes a react element title via props", () => {
    wrapper.setProps({ title: <IconHelp /> });
    expect(wrapper).toMatchSnapshot();
    expect(IconHelp).toBeDefined();
  });

  it("passes a title color via props", () => {
    wrapper.setProps({ titleColor: theme.colors.base });
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper.find("collapsible__ToggleWrap").find("typography__H4")
    ).toHaveStyleRule("color", theme.colors.base);
  });

  it("passes a title hover color via props", () => {
    wrapper.setProps({ titleHoverColor: theme.colors.muted });
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper.find("collapsible__ToggleWrap").props().titleHoverColor
    ).toBe(theme.colors.muted);
    // expect(wrapper.find('collapsible__ToggleWrap')).toHaveStyleRule('color', theme.colors.muted, { modifier: '.c1:hover .c2' });
  });

  it("sets a style object on the component wrapper via props", () => {
    wrapper.setProps({ wrapperStyle: { margin: "8px" } });
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper.find("collapsible__ElementWrap").props().style
    ).toHaveProperty("margin", "8px");
  });
});
