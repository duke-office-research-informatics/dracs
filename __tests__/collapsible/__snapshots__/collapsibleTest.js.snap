// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible matches snapshot 1`] = `
.c5 {
  fill: #b5b5b5;
  vertical-align: top;
}

.c7 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 8px;
}

.c3 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
  color: #0680cd;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1:hover .c2 {
  color: #53a4dc;
}

.c4 {
  cursor: pointer;
}

.c6 {
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h4
      className="c2 c3"
    >
      hello
    </h4>
    <button
      className="c4"
      onClick={[Function]}
    >
      <svg
        aria-label="Down Icon"
        className="c5"
        color="#b5b5b5"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          Down Icon
        </title>
        <desc>
          downward facing caret
        </desc>
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
  <div
    className="c6"
  >
    <svg
      aria-label="open customize and profile menu"
      className="c7"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;

exports[`Collapsible matches snapshot and returns no a11y violations when active is set to true 1`] = `
.c5 {
  fill: #b5b5b5;
  vertical-align: top;
}

.c7 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 8px;
}

.c3 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
  color: #0680cd;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1:hover .c2 {
  color: #53a4dc;
}

.c4 {
  cursor: pointer;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h4
      className="c2 c3"
    >
      hello
    </h4>
    <button
      className="c4"
      onClick={[Function]}
    >
      <svg
        aria-label="send icon"
        className="c5"
        color="#b5b5b5"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          send icon
        </title>
        <path
          d="M12 8l-6 6 1.41 1.41L12 10.83l4.59 4.58L18 14z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
  <div
    className="c6"
  >
    <svg
      aria-label="open customize and profile menu"
      className="c7"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;

exports[`Collapsible passes a react element title via props 1`] = `
.c6 {
  fill: #b5b5b5;
  vertical-align: top;
}

.c8 {
  fill: #fff;
  vertical-align: top;
}

.c4 {
  fill: #666;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 8px;
}

.c3 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
  color: #0680cd;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1:hover .c2 {
  color: #53a4dc;
}

.c5 {
  cursor: pointer;
}

.c7 {
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h4
      className="c2 c3"
    >
      <svg
        aria-label="Help Icon"
        className="c4"
        color="#666"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          Help Icon
        </title>
        <desc>
          qustion mark
        </desc>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
        />
      </svg>
    </h4>
    <button
      className="c5"
      onClick={[Function]}
    >
      <svg
        aria-label="Down Icon"
        className="c6"
        color="#b5b5b5"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          Down Icon
        </title>
        <desc>
          downward facing caret
        </desc>
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
  <div
    className="c7"
  >
    <svg
      aria-label="open customize and profile menu"
      className="c8"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;

exports[`Collapsible passes a title color via props 1`] = `
.c5 {
  fill: #b5b5b5;
  vertical-align: top;
}

.c7 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 8px;
}

.c3 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
  color: #131516;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1:hover .c2 {
  color: #53a4dc;
}

.c4 {
  cursor: pointer;
}

.c6 {
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h4
      className="c2 c3"
    >
      hello
    </h4>
    <button
      className="c4"
      onClick={[Function]}
    >
      <svg
        aria-label="Down Icon"
        className="c5"
        color="#b5b5b5"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          Down Icon
        </title>
        <desc>
          downward facing caret
        </desc>
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
  <div
    className="c6"
  >
    <svg
      aria-label="open customize and profile menu"
      className="c7"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;

exports[`Collapsible passes a title hover color via props 1`] = `
.c5 {
  fill: #b5b5b5;
  vertical-align: top;
}

.c7 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 8px;
}

.c3 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
  color: #0680cd;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.jfLYNq:hover .c2 {
  color: #53a4dc;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1:hover .c2 {
  color: #696969;
}

.c4 {
  cursor: pointer;
}

.c6 {
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h4
      className="c2 c3"
    >
      hello
    </h4>
    <button
      className="c4"
      onClick={[Function]}
    >
      <svg
        aria-label="Down Icon"
        className="c5"
        color="#b5b5b5"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          Down Icon
        </title>
        <desc>
          downward facing caret
        </desc>
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
  <div
    className="c6"
  >
    <svg
      aria-label="open customize and profile menu"
      className="c7"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;

exports[`Collapsible passes click function to toggle via onToggleClick prop 1`] = `
.c5 {
  fill: #b5b5b5;
  vertical-align: top;
}

.c7 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 8px;
}

.c3 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
  color: #0680cd;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1:hover .c2 {
  color: #53a4dc;
}

.c4 {
  cursor: pointer;
}

.c6 {
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h4
      className="c2 c3"
    >
      hello
    </h4>
    <button
      className="c4"
      onClick={[Function]}
    >
      <svg
        aria-label="Down Icon"
        className="c5"
        color="#b5b5b5"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          Down Icon
        </title>
        <desc>
          downward facing caret
        </desc>
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
  <div
    className="c6"
  >
    <svg
      aria-label="open customize and profile menu"
      className="c7"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;

exports[`Collapsible sets a style object on the component wrapper via props 1`] = `
.c5 {
  fill: #b5b5b5;
  vertical-align: top;
}

.c7 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 8px;
}

.c3 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
  color: #0680cd;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1:hover .c2 {
  color: #53a4dc;
}

.kJjZNS:hover .c2 {
  color: #696969;
}

.c4 {
  cursor: pointer;
}

.c6 {
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
  style={
    Object {
      "margin": "8px",
    }
  }
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h4
      className="c2 c3"
    >
      hello
    </h4>
    <button
      className="c4"
      onClick={[Function]}
    >
      <svg
        aria-label="Down Icon"
        className="c5"
        color="#b5b5b5"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          Down Icon
        </title>
        <desc>
          downward facing caret
        </desc>
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
  <div
    className="c6"
  >
    <svg
      aria-label="open customize and profile menu"
      className="c7"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;

exports[`Collapsible sets wrapper padding via props 1`] = `
.c5 {
  fill: #b5b5b5;
  vertical-align: top;
}

.c7 {
  fill: #fff;
  vertical-align: top;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 8px;
}

.c3 {
  font-family: "Roboto","Open-Sans",Sans-Serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #545454;
  font-weight: 400;
  margin: 4px 0;
  padding: 0;
  color: #0680cd;
  text-transform: uppercase;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  -webkit-letter-spacing: 0.025em;
  -moz-letter-spacing: 0.025em;
  -ms-letter-spacing: 0.025em;
  letter-spacing: 0.025em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c1:hover .c2 {
  color: #53a4dc;
}

.c4 {
  cursor: pointer;
}

.c6 {
  display: none;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <h4
      className="c2 c3"
    >
      hello
    </h4>
    <button
      className="c4"
      onClick={[Function]}
    >
      <svg
        aria-label="Down Icon"
        className="c5"
        color="#b5b5b5"
        height={24}
        role="img"
        viewBox="0 0 24 24"
        width={24}
      >
        <title>
          Down Icon
        </title>
        <desc>
          downward facing caret
        </desc>
        <path
          d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </button>
  </div>
  <div
    className="c6"
  >
    <svg
      aria-label="open customize and profile menu"
      className="c7"
      color="#fff"
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
    >
      <title>
        menu icon
      </title>
      <desc>
        hamburger menu icon
      </desc>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </div>
</div>
`;
